src/main/java/com/ldtteam/structurize/placement/handlers/entity/EntityHandlers.java

```diff
@@ -0,0 +1,226 @@
+package com.ldtteam.structurize.placement.handlers.entity;
+
+import com.ldtteam.structurize.placement.structure.IStructureHandler;
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.Display;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.Mob;
+import net.minecraft.world.entity.decoration.ArmorStand;
+import net.minecraft.world.entity.decoration.HangingEntity;
+import net.minecraft.world.entity.decoration.ItemFrame;
+import net.minecraft.world.entity.decoration.Painting;
+import net.minecraft.world.entity.vehicle.ContainerEntity;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.item.SpawnEggItem;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.common.ForgeSpawnEggItem;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class containing all Entity Handler implementations.
+ */
+public final class EntityHandlers
+{
+    public static final List<IEntityHandler> handlers = new ArrayList<>();
+    static
+    {
+        handlers.add(new MobEntityHandler());
+        handlers.add(new TextDisplayHandler());
+        handlers.add(new ItemFrameHandler());
+        handlers.add(new HangingEntityHandler());
+        handlers.add(new ArmorStandHandler());
+        handlers.add(new ContainerEntityHandler());
+
+        handlers.add(new DefaultEntityHandler());   // always last
+    }
+
+    /**
+     * Allows for adding new handlers without having to clear the list
+     * in other mods just to override one
+     * @param handler the new handler to add
+     * @param override the class to override if it can be found
+     */
+    public static void add(final IEntityHandler handler, final Class<?> override)
+    {
+        for (int i = 0; i < handlers.size(); i++)
+        {
+            if (override.isInstance(handlers.get(i)))
+            {
+                handlers.set(i, handler);
+                return;
+            }
+        }
+        add(handler);
+    }
+
+    /**
+     * Adds a handler to the start of the handlers list,
+     * effectively overriding existing ones with similar
+     * 'canHandle' functions because this one will evaluate before them
+     * @param handler the new handler to add
+     */
+    public static void add(final IEntityHandler handler)
+    {
+        handlers.add(1, handler);
+    }
+
+    /**
+     * Private constructor to hide implicit one.
+     */
+    private EntityHandlers()
+    {
+        // Intentionally left empty.
+    }
+
+    /**
+     * Handler for all kinds of mob
+     */
+    public static class MobEntityHandler implements IEntityHandler
+    {
+        @Override
+        public boolean canHandle(final Entity entity)
+        {
+            return entity instanceof Mob;

```

Comment:
don't have ide open, what is lower livingEntity or mob in entity class hierarchy? iirc mob implements livingentity, if so why not instanceof livingEntity?
(i see it was in old code, just asking)

Reply:
Mob is derived from LivingEntity.  There's probably no particular reason this couldn't be LivingEntity, but I was trying to keep previous behaviour, and that's what it did before.

Reply:
Makes sense, wondering if we want to resolve the rest of the set - ie. complement to Mob in LivingEntity set